src/components/gltf-model-plus.js

```diff
@@ -341,127 +340,221 @@ function runMigration(version, json) {
+            if (node) {
+              track.name = track.name.replace(/^[^.]*\./, node.uuid + ".");
+              break;
+            }
+          }
+        }
       }
+    }
 
-      node.extras.gltfIndex = i;
+    //
+    if (this.fileMap) {
+      // The GLTF is now cached as a THREE object, we can get rid of the original blobs
+      Object.keys(this.fileMap).forEach(URL.revokeObjectURL);
     }
+
+    //
+    gltf.scene.animations = gltf.animations;
   }
+}
 
-  // Mark the special nodes/meshes in json for efficient parse, all json manipulation should happen before this point
-  parser.markDefs();
+class GLTFHubsLightMapExtension {
+  constructor(parser) {
+    this.parser = parser;
+    this.name = "MOZ_lightmap";
+  }
+
+  // @TODO: Ideally we should use extendMaterialParams hook.
+  //        But the current official glTF loader doesn't fire extendMaterialParams
+  //        hook for unlit and specular-glossiness materials.
+  //        So using loadMaterial hook as workaround so far.
+  //        Cons is loadMaterial hook is fired as _invokeOne so
+  //        if other plugins defining loadMaterial is registered
+  //        there is a chance that this light map extension handler isn't called.
+  //        The glTF loader should be updated to remove the limitation.
+  loadMaterial(materialIndex) {
+    const parser = this.parser;
+    const json = parser.json;
+    const materialDef = json.materials[materialIndex];
+
+    if (!materialDef.extensions || !materialDef.extensions[this.name]) {
+      return null;
+    }
 
-  const materials = parser.json.materials;
-  const extensionDeps = [];
-  if (materials) {
-    for (let i = 0; i < materials.length; i++) {
-      const materialNode = materials[i];
+    const extensionDef = materialDef.extensions[this.name];
 
-      if (!materialNode.extensions) continue;
+    const pending = [];
 
-      if (materialNode.extensions.MOZ_lightmap) {
-        extensionDeps.push(loadLightmap(parser, i));
-      }
+    pending.push(parser.loadMaterial(materialIndex));
+    pending.push(parser.getDependency("texture", extensionDef.index));
+
+    return Promise.all(pending).then(results => {
+      const material = results[0];
+      const lightMap = results[1];
+      material.lightMap = lightMap;
+      material.lightMapIntensity = extensionDef.intensity !== undefined ? extensionDef.intensity : 1;
+      return material;
+    });
+  }
+}
+
+class GLTFHubsTextureBasisExtension {
+  constructor(parser) {
+    this.parser = parser;
+    this.name = "MOZ_HUBS_texture_basis";
+  }
+
+  loadTexture(textureIndex) {
+    const parser = this.parser;
+    const json = parser.json;
+    const textureDef = json.textures[textureIndex];
+
+    if (!textureDef.extensions || !textureDef.extensions[this.name]) {
+      return null;
+    }
+
+    if (!parser.options.ktx2Loader) {
+      // @TODO: Display warning (only if the extension is in extensionsRequired)?
```

Comment:
In practice we will always be providing a ktx2loader so if this happens its certainly a bug... We probably don't want to fail silently.

Reply:
OK, I would add warning here